src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedStaticConstructor.cs

```diff
@@ -371,5 +371,56 @@ internal override int CalculateLocalSyntaxOffset(int localPosition, SyntaxTree l
             var containingType = (SourceMemberContainerTypeSymbol)this.ContainingType;
             return containingType.CalculateSyntaxOffsetInSynthesizedConstructor(localPosition, localTree, isStatic: true);
         }
+
+#nullable enable
+        private ThreeState _shouldEmit = ThreeState.Unknown;
+        internal bool ShouldEmit(ImmutableArray<BoundInitializer> boundInitializersOpt = default)
+        {
+            if (_shouldEmit.HasValue())
+            {
+                return _shouldEmit.Value();
+            }
+
+            var shouldEmit = InitializeShouldEmit(boundInitializersOpt);
+            _shouldEmit = shouldEmit.ToThreeState();
+            return shouldEmit;
+        }
+
+        private bool InitializeShouldEmit(ImmutableArray<BoundInitializer> boundInitializersOpt = default)
+        {
+            if (boundInitializersOpt.IsDefault)
+            {
+                if (!(ContainingType is SourceMemberContainerTypeSymbol sourceType))
+                {
+                    return true;
+                }
+
+                var unusedDiagnostics = DiagnosticBag.GetInstance();
+                boundInitializersOpt = Binder.BindFieldInitializers(
+                    DeclaringCompilation,
+                    sourceType.IsScriptClass ? sourceType.GetScriptInitializer() : null,
+                    sourceType.StaticInitializers,
+                    unusedDiagnostics,
+                    out _);
+                unusedDiagnostics.Free();
+            }
+
+            foreach (var initializer in boundInitializersOpt)
+            {
+                if (!(initializer is BoundFieldEqualsValue { Value: { } value }))
+                {
+                    // this isn't a BoundFieldEqualsValue, so this initializer is doing
+                    // something we don't understand. Better just emit it.
+                    return true;
+                }
+
+                if (!value.IsDefaultValue())
```

Comment:
>if (!value.IsDefaultValue()) [](start = 16, length = 28)

Stopping at handling just these simple cases is an option and we certainly don't have to expand it in this PR. This is more a design question and I think it is worth having an offline discussion at some point.